modules/imgproc/src/fixedpoint.inl.hpp

```diff
@@ -67,33 +69,37 @@ class fixedpoint64
         uint64_t val0_h = (sh2   & 0xFFFFFFFF) + (sh1_0 >> 32) + (sh1_1 >> 32) + (val0_l >> 32);
         val0_l &= 0xFFFFFFFF;
 
-        if ( (sh2 >> 32) || (val0_h >> ressign ? 32 : 31) )
-            return (ressign ? ~(int64_t)0x7FFFFFFFFFFFFFFF : (int64_t)0x7FFFFFFFFFFFFFFF);
+        if (sh2 > CV_BIG_INT(0x7FFFFFFF) || val0_h > CV_BIG_INT(0x7FFFFFFF))
```

Comment:
Shouldn't it be `if (sh2 > CV_BIG_UINT(0xFFFFFFFF) || val0_h > (ressign ? CV_BIG_UINT(0x80000000) : CV_BIG_INT(0x7FFFFFFF)))`